### DISCUSSION TITLE
Lesson 7 - testWithdrawWithASingleFunder - gas not used?

### DISCUSSION THREAD

**Discussion by tomoglava** [Cyfrin/foundry-full-course-cu#811] [Q&A] [answered]
Hello,

so function goes like this:
```
function testWithdrawWithASingleFunder() public funded {
        //Arrange
        uint256 startingOwnerBalance = fundMe.getOwner().balance;
        uint256 startingFundMeBalance = address(fundMe).balance;

        //Act
        vm.prank(fundMe.getOwner());
        fundMe.withdraw();

        //Assert
        uint256 endingOwnerBalance = fundMe.getOwner().balance;
        uint256 endingFundMeBalance = address(fundMe).balance;

        assertEq(endingFundMeBalance, 0);
        assertEq(startingOwnerBalance + startingFundMeBalance, endingOwnerBalance);
    }
```
How come this passes test since `fundMe.withdraw()` should cost some gas to Owner and thus it must be: 
`(startingOwnerBalance - gasUsed) + startingFundMeBalance  = endingOwnerBalance`?

Also digression: modifier `funded` seems unnecessary here, doesn't it?

Thnx in advance for answers.

**Comment by theirrationalone** [comment, marked as answer]
Follow Patrick thoroughly.... You're very close to get that Gas Stuff too. Here, We're doing similar things as of `Patrick's fcc HH` but foundry has some differences.

**Reply by tomoglava** [reply]
I see it's about half an hour latter explained.
